user_vocab.py

```diff
@@ -1,21 +1,47 @@
 import os.path
 
-
 filename = "kindle_highlights.txt"
 
+
+#checks file exists
 if not os.path.isfile(filename):
-    print('File does not exist.')
-    quit()
-else:
-    with open(filename) as f:
-        content = f.read().splitlines()
+    raise Exception('File does not exist.')
+
+
+#checks file is of supported type (.txt only at present)
+if filename.endswith('.txt'):
+    pass
+else:  
+    raise Exception("user_vocab parsing curently unsupported file type")
+
+    
+with open(filename) as f:
+
+    content = f.read().splitlines()
 
+#removes any characters that are not letters ('abc')
 def rmv_non_charac(word):
```

Comment:
```suggestion
def remove_non_alpha_characters(word):
```

I found your missing vowels :P

Also - all possible components of a string are "characters" inlcuding "non-printing characters" and "control characters". https://en.wikipedia.org/wiki/Control_character

The set `[a-zA-Z]` is the [latin] alphabet commonly shortened to alpha. When including numbers `[a-zA-Z0-9]` we say alphanumeric.